### PR TITLE
viz: add a new uninstall command

### PR DESCRIPTION
**Pull request by Pothulapati**
Depends on #3340

This adds a new `linkerd viz uninstall` command emitting the resources
with the `linkerd.io/extension=linkerd-viz` label set.

This is similar to that of the other linkerd uninstall commands.

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

### CODE REVIEW

File: viz/cmd/uninstall.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+func newCmdUninstall() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "uninstall",
+		Args:  cobra.NoArgs,
+		Short: "Output Kubernetes resources to uninstall the Linkerd viz extension",
+		Long: `Output Kubernetes resources to uninstall the Linkerd viz extension.
+
+This command provides all Kubernetes namespace-scoped and cluster-scoped resources (e.g services, deployments, RBACs, etc.) necessary to uninstall the Linkerd-viz extension.`,
+		Example: `linkerd uninstall | kubectl delete -f -`,
```
Comment (alpeb) [inline viz/cmd/uninstall.go:22]:
```suggestion
		Example: `linkerd viz uninstall | kubectl delete -f -`,
```

File: viz/cmd/uninstall.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func newCmdUninstall() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "uninstall",
+		Args:  cobra.NoArgs,
+		Short: "Output Kubernetes resources to uninstall the Linkerd viz extension",
+		Long: `Output Kubernetes resources to uninstall the Linkerd viz extension.
```
Comment (alpeb) [inline viz/cmd/uninstall.go:19]: This suggestion avoids some discrepancy in naming:
```
$ linkerd viz
...
Available Commands:
  install     Output Kubernetes resources to install linkerd-viz extension
  uninstall   Output Kubernetes resources to uninstall the Linkerd viz extension
```

```suggestion
		Short: "Output Kubernetes resources to uninstall the linkerd-viz extension",
		Long: `Output Kubernetes resources to uninstall the linkerd-viz extension.
```

**Review by alpeb — APPROVED** [review]
LGTM, just some nits :+1:

**Review by adleong — COMMENTED** [review]
LGTM pending @alpeb's suggestions.

File: viz/cmd/uninstall.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+func uninstallRunE(ctx context.Context) error {
+	k8sAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 0)
+	if err != nil {
+		return err
+	}
+
+	resources, err := resource.FetchKubernetesResources(ctx, k8sAPI,
+		metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", k8s.LinkerdExtensionLabel, "linkerd-viz")},
```
Comment (kleimkuhler) [inline viz/cmd/uninstall.go:38]:
```suggestion
		metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", k8s.LinkerdExtensionLabel, defaultVizNamespace)},
```

**Review by kleimkuhler — APPROVED** [review]
Small nit, otherwise LGTM 👍

**Pull request merged**